test_unit/test_sink.cpp

```diff
@@ -40,6 +40,28 @@ using namespace g3;
    EXPECT_TRUE(1 == count->load());
 }
 
+TEST(Sink, OneSinkRemove) {
+using namespace g3;
+   AtomicBoolPtr flag = make_shared < atomic<bool >> (false);
+   AtomicIntPtr count = make_shared < atomic<int >> (0);
+   {
+      auto worker = g3::LogWorker::createLogWorker();
+      auto handle = worker->addSink(std2::make_unique<ScopedSetTrue>(flag, count), &ScopedSetTrue::ReceiveMsg);
+      EXPECT_FALSE(flag->load());
+      EXPECT_TRUE(0 == count->load());
+
+      LogMessagePtr message{std2::make_unique<LogMessage>("test", 0, "test", DEBUG)};
+      message.get()->write().append("this message should trigger an atomic increment at the sink");
+      worker->save(message);
+      worker->removeSink(std::move(handle));
+      EXPECT_TRUE(flag->load());
+      EXPECT_TRUE(1 == count->load());
+
+      // TODO: saving message after sink was removed will cause seg fault in worker destructor
```

Comment:
This comment is bothering me. It should be perfectly fine to write log messages when the sink is removed.
What is causing the seg fault?